js-envs/src/main/scala/org/scalajs/jsenv/Utils.scala

```diff
@@ -0,0 +1,30 @@
+/*                     __                                               *\
+**     ________ ___   / /  ___      __ ____  Scala.js JS environments   **
+**    / __/ __// _ | / /  / _ | __ / // __/  (c) 2013, LAMP/EPFL        **
+**  __\ \/ /__/ __ |/ /__/ __ |/_// /_\ \    http://scala-js.org/       **
+** /____/\___/_/ |_/____/_/ | |__/ /____/                               **
+**                          |/____/                                     **
+\*                                                                      */
+
+
+package org.scalajs.jsenv
+
+import scala.concurrent.duration._
+
+private[jsenv] object Utils {
+  final class OptDeadline private (
+      val deadline: Deadline /* nullable */) extends AnyVal {
+    def millisLeft: Int =
+      if (deadline == null) 0
+      else (deadline.timeLeft.toMillis.toInt max 1)
```

Comment:
Because 0 means Infinite. If the deadline is present but overdue, the best approximation we can have is 1 ms, not Infinite.